cdc/owner/ddl_sink.go

```diff
@@ -144,12 +149,16 @@ func (s *ddlSinkImpl) run(ctx cdcContext.Context, id model.ChangeFeedID, info *m
 				ctx.Throw(err)
 				return
 			case <-ticker.C:
-				checkpointTs := atomic.LoadUint64(&s.checkpointTs)
+				s.mu.Lock()
+				checkpointTs := s.mu.checkpointTs
 				if checkpointTs == 0 || checkpointTs <= lastCheckpointTs {
+					s.mu.Unlock()
 					continue
 				}
+				tables := s.mu.currentTableNames
+				s.mu.Unlock()
 				lastCheckpointTs = checkpointTs
-				if err := s.sink.EmitCheckpointTs(ctx, checkpointTs); err != nil {
+				if err := s.sink.EmitCheckpointTs(ctx, checkpointTs, tables); err != nil {
```

Comment:
Is it possible that the tablesName hold by `tables` is already stale? eg. New tables are added or existing tables are dropped.

Reply:
It will be possible, but it is unavoidable? Because we are asynchronous flush.
So we can only update yes tables with checkpoint ts at that time as much as possible?

Reply:
Also we are emitting checkpoint ts first and then executing ddl. so the problem is not too serious. However, there is still a possibility of inconsistency due to asynchronous reasons. For example, a new table is added within 1 second. But I think it takes time for ddl to be distributed in sink, so all we can do is to use the current tables as much as possible?

Reply:
> Also we are emitting checkpoint ts first and then executing ddl. so the problem is not too serious. However, there is still a possibility of inconsistency due to asynchronous reasons. For example, a new table is added within 1 second. But I think it takes time for ddl to be distributed in sink, so all we can do is to use the current tables as much as possible?

If missing some checkpointTs for those new tables is tolerant by downstream, we don't have to maintain the latest tableNames. BTW, I'm wondering how is the checkpointTs handled by downstream?

Reply:
It will be used to sort with commit ts.

Reply:
> If missing some checkpointTs for those new tables is tolerant by downstream

Theoretically we shouldn't emit checkpoint ts before ddl broadcasts to kafka, so it only sends more, not less?